File: backend/schemas/face.py
from datetime import datetime

from pydantic import BaseModel, Field


class PhotoFaceBrief(BaseModel):
    id: int
    photo_id: int
    face_cluster_id: int | None = None
    bbox_x: int
    bbox_y: int
    bbox_w: int
    bbox_h: int
    confidence: float
    thumbnail_path: str | None = None

    model_config = {"from_attributes": True}


class FaceClusterResponse(BaseModel):
    id: int
    name: str | None = None
    representative_face_id: int | None = None
    face_count: int = 0
    cover_thumbnail: str | None = None
    created_at: datetime | None = None

    model_config = {"from_attributes": True}


class FaceClusterDetailResponse(FaceClusterResponse):
    faces: list[PhotoFaceBrief] = []


class FaceClusterListResponse(BaseModel):
    items: list[FaceClusterResponse]
    total: int


class MergeClusterRequest(BaseModel):
    cluster_ids: list[int] = Field(min_length=2)


class UpdateClusterRequest(BaseModel):
    name: str = Field(min_length=1, max_length=64)

File: frontend/src/types/face.ts
export interface FaceCluster {
  id: number
  name: string | null
  representative_face_id: number | null
  face_count: number
  cover_thumbnail: string | null
  created_at: string | null
}

export interface PhotoFace {
  id: number
  photo_id: number
  face_cluster_id: number | null
  bbox_x: number
  bbox_y: number
  bbox_w: number
  bbox_h: number
  confidence: number
  thumbnail_path: string | null
}

export interface FaceClusterDetail extends FaceCluster {
  faces: PhotoFace[]
}

export interface FaceClusterListResponse {
  items: FaceCluster[]
  total: number
}
